packages/hdwallet-core/src/bitcoin.ts

```diff
@@ -37,63 +47,179 @@ export interface BitcoinOutput {
+export interface BitcoinTxBase {
   version: number;
   locktime: number;
   vin: Array<BitcoinInput>;
   vout: Array<BitcoinOutput>;
-
-  type?: number; // Dash
-  extraPayload?: string; // Dash
-  extraPayloadSize?: number; // Dash
 }
 
+type BitcoinTxDIP2 = BitcoinTxBase & {
+  type: number;
+  extraPayload: string;
+};
+
+export type BitcoinTx = GuardedUnion<BitcoinTxBase | BitcoinTxDIP2>;
+
 /**
  * Input for a transaction we're about to sign.
  */
-export interface BTCSignTxInput {
-  /** bip32 path to sign the input with */
+type BTCSignTxInputBase = {
+  vout: number;
   addressNList: BIP32Path;
-  scriptType?: BTCInputScriptType;
-  sequence?: number;
-  amount: string;
+  amount?: string;
+};
+
+type BTCSignTxInputNativeBase = BTCSignTxInputBase & {
+  txid: string;
+};
+
+type BTCSignTxInputNativeSegwitBase = BTCSignTxInputNativeBase & {
+  scriptType: BTCInputScriptType.SpendWitness | BTCInputScriptType.SpendP2SHWitness;
+};
+
+type BTCSignTxInputNativeSegwitWithHex = BTCSignTxInputNativeSegwitBase & {
+  hex: string;
+};
+
+type BTCSignTxInputNativeSegwitWithTx = BTCSignTxInputNativeSegwitBase & {
+  tx: BitcoinTx;
   vout: number;
+  amount: string;
+};
+
+type BTCSignTxInputNativeSegwit = BTCSignTxInputNativeSegwitWithHex | BTCSignTxInputNativeSegwitWithTx;
+
+type BTCSignTxInputNativeNonSegwit = BTCSignTxInputNativeBase & {
+  scriptType: Exclude<BTCInputScriptType, BTCSignTxInputNativeSegwit["scriptType"]>;
+  hex: string;
+};
+
+type BTCSignTxInputNativeUnguarded = BTCSignTxInputNativeSegwit | BTCSignTxInputNativeNonSegwit;
+export type BTCSignTxInputNative = GuardedUnion<BTCSignTxInputNativeUnguarded>;
+
+type BTCSignTxInputKKBase = BTCSignTxInputBase & {
   txid: string;
-  tx?: BitcoinTx; // Required for p2sh, not required for segwit
+  amount: string;
+  sequence?: number;
+};
+
+type BTCSignTxInputKKSegwit = BTCSignTxInputKKBase & {
+  scriptType: BTCInputScriptType.SpendWitness | BTCInputScriptType.SpendP2SHWitness | BTCInputScriptType.External;
+};
+
+type BTCSignTxInputKKNonSegwit = BTCSignTxInputKKBase & {
+  scriptType: Exclude<BTCInputScriptType, BTCSignTxInputKKSegwit["scriptType"]>;
+  tx: BitcoinTx;
+};
+
+type BTCSignTxInputKKUnguarded = BTCSignTxInputKKNonSegwit | BTCSignTxInputKKSegwit;
+export type BTCSignTxInputKK = GuardedUnion<BTCSignTxInputKKUnguarded>;
+
+export type BTCSignTxInputTrezor = BTCSignTxInputBase & {
+  txid: string;
+  amount: string;
+  scriptType: BTCInputScriptType;
+};
+
+export type BTCSignTxInputLedger = BTCSignTxInputBase & {
+  addressNList: BIP32Path;
+  scriptType: BTCInputScriptType;
   hex: string;
-  type?: number; // Dash
-  extraPayloadSize?: number; // Dash
-  extraPayload?: string; // Dash
-}
+};
+
+export type BTCSignTxInput = BTCSignTxInputNative & BTCSignTxInputKK & BTCSignTxInputTrezor & BTCSignTxInputLedger;
+export type BTCSignTxInputUnguarded = BTCSignTxInputNativeUnguarded & BTCSignTxInputKKUnguarded & BTCSignTxInputTrezor & BTCSignTxInputLedger;
```

Comment:
This dichotomy is dumb. We should fix all btcSignTx implementations to use a single representation of the input tx -- probably `hex` -- and eliminate the confusion.

Of course there are probably edge-cases in clients which are passing in inconsistent `tx` and `hex` values -- so normalizing how we actually process the data, rather than just how we type it, may change client behavior and will need a deft hand.